### PR TITLE
Adding all the rule states in getTapRecallUsage and a bug fix

### LINKED ISSUE #8736: report of quota used by crab_input is wrong
this code can not work for crab_input account.  I am not sure if it is currently used to populate grafana panel or not. If yes I am very surprised that it give a sensible 21PB. I expect that most rules for crab_input are OK, not REPLICATING/SUSPENDED/STUCK 
https://github.com/dmwm/CRABServer/blob/98bdaa7e08045c771cf961d5364be53ee67e7819/src/python/RucioUtils.py#L117-L128

### PR DESCRIPTION
**Pull request by aspiringmind-code**
Fix #8736

### CODE REVIEW

File: src/python/RucioUtils.py
Comment (belforte) [inline src/python/RucioUtils.py:127]: if you want to select all possible states, why not simply remove this line ?
In any case, although we use automatic approve here, rules in WAITING_APPROVAL should not be counted as "using space". I am not sure about rules in INJECTING, i.e. whether it comes before or after APPROVAL.

**Comment by belforte** [comment]
Aside the minor comment above, there is a fundamental problem here.

* for tape recall we do not want to count rules in OK status, since we want to know what is being asked, not what is on disk. True that now we remove OK rules after 4 days, but still it is better to avoid it
* for input lock almost everything should be OK

I am still puzzled that code could find  a sensible value for input lock while ignoring OK rules. That needs to be understood.
Can you look at it, or do I need to do it ?

**Review by belforte — CHANGES_REQUESTED** [review]

File: src/python/RucioUtils.py
Comment (aspiringmind-code) [reply src/python/RucioUtils.py:127]: I didn't remove the line as i still wanted to test that only one of these 6 states are being used and not any random value. Acc to Hasan, INJECT is the first step and it comes before WAITING_APPROVAL, followed by REPLICATING.

**Comment by aspiringmind-code** [comment]
I will look at it.

File: src/python/RucioUtils.py
Comment (belforte) [reply src/python/RucioUtils.py:127]: So we should not count INJECT either. Or account separately for "pendind rules". But let's try not to make it too complicated

**Comment by aspiringmind-code** [comment]
@belforte The use of getTapeRecallUsage was merged by me on 17th Sept in https://github.com/dmwm/CRABServer/pull/8697 and the release we are running is that of v3.240904 (4th September). Hence the changes are not yet in production and therefore we have the correct thing from using rucioClient.get_local_account_usage in the 4th Sept version.

**Comment by aspiringmind-code** [comment]
We did the change to address https://github.com/dmwm/CRABServer/issues/8661 as we wanted to filter by activity=AnalysisTapeRecall instead of user, and use user argument only if provided.

**Comment by belforte** [comment]
thanks for fixing my confusion ! and sorry for noise.

**Comment by belforte** [comment]
@aspiringmind-code I am a bit lost here, sorry. Are you waiting for my feedback ?

**Comment by aspiringmind-code** [comment]
@belforte Ah, no...this is not complete. I'll fix it and tag you for your feedback when done.

**Review by belforte — CHANGES_REQUESTED** [review]

File: src/python/RucioUtils.py
Comment (belforte) [reply src/python/RucioUtils.py:119]: yes OK

**Pull request merged**
